car-pool/car-mirror/http.md

```diff
@@ -63,45 +70,59 @@ POST /api/v0/dag/push?stream={bool}&diff={ipns | dnslink | cid}
 
 ### 2.1.1 `stream` Parameter
 
-This field specifies if the Provider should use discrete or streaming. `stream` MUST default to `false`. If streaming is set to `true`, the response MUST be a streaming CAR file and transmitted over [HTTP Streaming](https://datatracker.ietf.org/doc/html/rfc7540#section-5).
+This field specifies if the Provider should use discrete or streaming. `stream` MUST default to `false`. If streaming is set to `true`, the response MUST be a streaming CAR file and transmitted over [HTTP Streaming].
 
 ### 2.1.2 `diff` Parameter
 
 The `diff` field is OPTIONAL. It represents a related CID to the one being pushed, and MAY be an IPNS record, DNSLink, or CID. The complete URI MUST be provided.
 
 While a mutable pointer (IPNS or DNSLink) MAY be resolved to a CID before the request is constructed, it is RECOMMENDED that the pointer be given when available. This strategy gives the Provider more flexibility for tracking the last value that they've seen for that pointer, and conveys more about the Requestor's intention.
 
-This field is primarily useful for the [narrowing step](#312-narrowing), and especially during a [cold call](#3131-cold-calls). Since the Requestor does not have knowledge of everything in the Provider's store, this field provides a hint for the Provider of what (finite) context to include as part of the Bloom in their response.
+This field is primarily useful for the narrowing step, and especially during a cold call. Since the Requestor does not have knowledge of everything in the Provider's store, this field provides a hint for the Provider of what (finite) context to include as part of the Bloom in their response.
 
 This field MUST NOT be interpreted as a CID root being sent.
 
 ## 2.2 Requestor Payload
 
-```cddl
-payload = {
-  bk : uint,  ; Bloom filter hash count
-  bb : bytes, ; Bloom filter Binary
-  pl : car,   ; Data payload
+The requestor payload MUST be serialized as [CBOR].
+
+```ipldsch
+type ReqPayload struct {
+  sr [Link]  -- Incomplete subgraph roots
+  bk Integer -- Bloom filter hash count
+  bb Bytes   -- Bloom filter Binary
 }
 ```
 
-The data payload MUST be given as a [CARv1](https://ipld.io/specs/transport/car/carv1/). If the streaming flag was set in the URL, then the CAR MUST be a [streaming CARv1](https://ipld.io/specs/transport/car/carv1/#performance).
-
 ## 2.3 Provider Payload
 
-The requestor payload MUST be serialized as [CBOR](https://cbor.io/). This schema is given in [CDDL (RFC8601)](https://datatracker.ietf.org/doc/html/rfc8610).
-
-```cddl
-payload = {
-  sr : [* cid], ; Incomplete subgraph roots
-  bk : uint,    ; Bloom filter hash count
-  bb : bytes,   ; Bloom filter Binary
-}
+```ipldsch
+type ProvPayload struct {
+  bk Integer -- Bloom filter hash count
+  bb Bytes   -- Bloom filter Binary
+  pl CarFile -- Data payload
+} representation tuple
```

Comment:
This is probably the most controversial part of this spec. IPLD Schema has [representation strategies](https://ipld.io/docs/schemas/features/representation-strategies/#struct-tuple-representation) for strcuts (and a couple other types).

With the `map` representation, IPLD will default canonicalize the order of fields. We want to impose order, hence the `tuple` (array) representation. As a bonus, it's now ordered, so we don't strictly speaking need those keys. Better yet: if you deserialize this with an IPLD codec (e.g. `dag-cbor`), you'll parse out a struct with the fields on it.